### ISSUE TITLE
Unclear warning in ImpactFunc.check()

### ISSUE THREAD

**Issue by bguillod** [CLIMADA-project/climada_python#228]
If I define an impact function with mdd or paa different from 0 at intensity 0, I get the following warnings from `ImpactFunc.check()`:

```
climada.entity.impact_funcs.base - WARNING - For intensity = 0, mdd != 0 or paa != 0. Consider shifting the origin of the intensity scale. In impact.calc the impact is always null at intensity = 0.
```

To me this suggests that even if both mdd and paa are not 0 for intensity==0, impact.calc will return null to me. However, looking at the code, I do not see this condition applied anywhere. Am I overseeing something here?

Note that in my case, paa is set to 1 for all intensities, but mdd is actually 0 for intensity==0, so I actually expect 0 impact from intensity==0. However, this warning made me doubt whether I correctly understand the impact calculation for that case.

**Comment by ThomasRoosli** [comment]
My understanding is that this condition follows from the implementation of hazard.intensity as a sparse matrix.
In [this line](https://github.com/CLIMADA-project/climada_python/blob/ad94019286b8461bcf07635e44d76cc92b4656f5/climada/engine/impact.py#L871) the impact function is only applied to nonzero elements. Meaning if mdd and paa would both be non-zero at intensity==0, this would not be recognized in the implementation as the calculation is never initiated for all elements where hazard.intensity==0.
That is why Sam and I introduced this warning at one point.

**Comment by bguillod** [comment]
Ok, that makes sense. However, why raise this warning if only one of mdd or paa is non-zero? This is only an issue if both are non-0, right?

**Comment by emanuel-schmid** [comment]
fixed with PR #436
